cmd/ftl/main.go

```diff
@@ -81,7 +80,7 @@ func main() {
 	ctx = log.ContextWithLogger(ctx, logger)
 
 	config, err := projectconfig.LoadConfig(ctx, cli.ConfigFlag)
-	if err != nil && !errors.Is(err, os.ErrNotExist) {
+	if err != nil {
```

Comment:
The semantics are a little more involved than this will allow for:

1. If no config is passed on the CLI, we should try to load `ftl-project.toml` from the root of the repo, but if it doesn't exist that's okay.
2. If a config is passed explicitly on the command-line, and doesn't exist, this should error.

The simplest way to do this is probably to just check if `cli.ConfigFlag == ""`

Reply:
I'm probably missing something, but I tried these cases with the following results:

1. No config passed to CLI and `ftl-project.toml` exists in the git root: succeeds, has populated Config.
2. No config passed to CLI and `ftl-project.toml` doesn't exist in the git root: succeeds, has empty Config.
3. Config passed to CLI that doesn't exist: error.

Relevant code is in [ConfigPaths](https://github.com/TBD54566975/ftl/blob/e5594953734f0683f1583b58ef59502557957e05/common/projectconfig/projectconfig.go#L42) which returns the config passed to CLI if it exists, the `ftl-project.toml` in the git root if it exists, otherwise an empty slice. The subsequent call to `Merge` fails if a path doesn't exist, but it's never provided a nonexistent `ftl-project.toml` in the git root.

I went down the path of testing by mocking out `GetDefaultConfigPath()` but that got unwieldy for this change 😅

Reply:
Ah awesome, apologies I forgot about that code in `ConfigPaths`, you're 100% right! LGTM